cpp/src/umap/knn_graph/algo.cuh

```diff
@@ -68,8 +68,8 @@ inline void launcher(const raft::handle_t& handle,
 {
   cudaPointerAttributes attr;
   RAFT_CUDA_TRY(cudaPointerGetAttributes(&attr, inputsA.X));
-  float* ptr          = reinterpret_cast<float*>(attr.devicePointer);
-  bool data_on_device = ptr != nullptr;
+  float* ptr          = reinterpret_cast<float*>(attr.hostPointer);
```

Comment:
Do pointers allocated with `new` and `malloc` also have a valid `attr.hostPointer`?